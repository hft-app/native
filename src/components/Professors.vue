<template>
    <div>
        <nav>
            <div class="container">
                <label>
                    <fa-icon class="icon icon-prepend icon-search" icon="search"/>
                    <input v-model="filter" class="search" placeholder="Suchen...">
                </label>
            </div>
        </nav>

        <div class="wrapper">
            <div class="list">
                <div class="container">

                    <div class="professor article" v-for="professor in filteredProfessors" v-bind:key="professor.email">
                        <div class="title">{{professor.name}}</div>
                        <div class="data">
                            <div class="infos">
                                <div class="room icon-prepend icon-fw icon">{{professor.room}}</div>
                                <div class="time icon-prepend icon-fw icon">{{professor.time}}</div>
                            </div>

                            <div class="actions">
                                <a class="icon icon-phone green" :href="'tel:' +professor.phone">
                                    <fa-icon icon="phone"/>
                                </a>
                                <a class="icon icon-envelope blue" :href="'mailto:'+professor.email">
                                    <fa-icon icon="envelope"/>
                                </a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="note">
                <div class="container">
                    <p>Professorenverzeichnis</p>
                    <p>© <a href="http://www.hft-stuttgart.de/" target="_blank">Hochschule für Technik Stuttgart</a></p>
                </div>
            </div>
        </div>
    </div>
</template>
<script>
    import data from "../data/professors.json"

    export default {
        data() {
            return {
                professors: data,
                filter: ""
            }
        },

        computed: {
            filteredProfessors() {
                if (this.filter !== "") {
                    const filter = this.filter.toLowerCase();
                    return this.professors.filter(professor => professor.name.toLowerCase().indexOf(filter) !== -1);
                } else {
                    return this.professors
                }
            }
        }
    }
</script>
<style lang="scss" scoped>
    @import '../colors';

    .list {
        padding-top: 60px;
    }

    .professor {
        display: flex;
        flex-direction: column;
        padding: 8px 14px 12px;

        &.hidden {
            display: none;
        }

        .title {
            font-size: 1.2em;
            padding-bottom: 8px;
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
        }

        .data {
            display: flex;
            align-items: flex-end;
            justify-content: space-between;

            .actions {
                display: flex;

                a {
                    color: #FFF;
                    height: 2em;
                    width: 2em;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    border-radius: 1.2em;
                    margin-left: 0.5em;

                    &.lu-phone::before {
                        font-size: 1.2em;
                    }
                }
            }

            .infos {
                font-size: 0.8em;
                overflow: hidden;
                font-weight: 300;

                div {
                    text-overflow: ellipsis;
                    white-space: nowrap;
                    overflow: hidden;

                    &::before {
                        color: $primary;
                    }

                    & + div {
                        margin-top: 4px;
                    }
                }
            }
        }
    }
</style>
