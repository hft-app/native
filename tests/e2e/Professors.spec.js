jest.mock('../../src/data/professors.json', () => [
  {
    'name': 'Prof. Dr. Rick Sanchez',
    'phone': '+49 (0)711 1245 8901',
    'email': 'rick.sanchez@hft-stuttgart.de',
    'room': '3/232',
    'time': 'Donnerstag 11:20 - 12:20'
  },
  {
    'name': 'Prof. Dr.-Ing. Morty Smith',
    'phone': '+49 (0)711 1245 8910',
    'email': 'morty.smith@hft-stuttgart.de',
    'room': '1/201',
    'time': 'nach Vereinbarung'
  }
], {virtual: true});

import {config, shallowMount} from '@vue/test-utils'
import Professors from 'components/Professors.vue';

config.mocks['$t'] = str => str;

describe('Professors', () => {
  test('renders correctly', () => {
    const wrapper = shallowMount(Professors, {stubs: ['fa-icon']});
    expect(wrapper.element).toMatchSnapshot()
  });

  test('renders correctly - filters', async () => {
    const wrapper = shallowMount(Professors, {stubs: ['fa-icon']});

    wrapper.find('.search').setValue('sAnchez');
    await wrapper.vm.$nextTick();

    expect(wrapper.element).toMatchSnapshot()
  });

  test('renders correctly - filters nothing found', async () => {
    const wrapper = shallowMount(Professors, {stubs: ['fa-icon']});

    wrapper.find('.search').setValue('dfg');
    await wrapper.vm.$nextTick();

    expect(wrapper.element).toMatchSnapshot()
  });
});



