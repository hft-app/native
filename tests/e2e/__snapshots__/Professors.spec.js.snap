// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Professors renders correctly - filters 1`] = `
<div
  class="wrapper"
>
  <nav>
    <div
      class="container"
    >
      <label>
        <fa-icon-stub
          class="icon icon-prepend icon-search"
          icon="search"
        />
         
        <input
          class="search"
          placeholder="page.professors.search"
        />
      </label>
    </div>
  </nav>
   
  <transition-stub
    mode="out-in"
    name="fade"
  >
    <div
      class="wrapper"
    >
      <div
        class="list"
      >
        <div
          class="container"
        >
          <transition-group-stub
            name="list-complete"
            tag="p"
          >
            <div
              class="professor article"
            >
              <div
                class="title"
              >
                Prof. Dr. Rick Sanchez
              </div>
               
              <div
                class="data"
              >
                <div
                  class="infos"
                >
                  <div>
                    <fa-icon-stub
                      class="room icon-prepend icon-fw icon"
                      icon="map-marker-alt"
                    />
                    
                    3/232
                  
                  </div>
                   
                  <div>
                    <fa-icon-stub
                      class="time icon-prepend icon-fw icon"
                      icon="calendar-alt"
                    />
                    
                    Donnerstag 11:20 - 12:20
                  
                  </div>
                </div>
                 
                <div
                  class="actions"
                >
                  <a
                    class="icon icon-phone green"
                    href="tel:+49 (0)711 1245 8901"
                  >
                    <fa-icon-stub
                      icon="phone"
                    />
                  </a>
                   
                  <a
                    class="icon icon-envelope blue"
                    href="mailto:rick.sanchez@hft-stuttgart.de"
                  >
                    <fa-icon-stub
                      icon="envelope"
                    />
                  </a>
                </div>
              </div>
            </div>
          </transition-group-stub>
        </div>
      </div>
       
      <div
        class="note"
      >
        <div
          class="container"
        >
          <p>
            page.professors.title
          </p>
           
          <p>
            
            © 
            <a
              href="http://www.hft-stuttgart.de/"
              target="_blank"
            >
              Hochschule für Technik Stuttgart
            </a>
          </p>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Professors renders correctly - filters nothing found 1`] = `
<div
  class="wrapper"
>
  <nav>
    <div
      class="container"
    >
      <label>
        <fa-icon-stub
          class="icon icon-prepend icon-search"
          icon="search"
        />
         
        <input
          class="search"
          placeholder="page.professors.search"
        />
      </label>
    </div>
  </nav>
   
  <transition-stub
    mode="out-in"
    name="fade"
  >
    <div
      class="screen container"
    >
      <fa-icon-stub
        class="icon-address-card-o icon"
        icon="address-card"
      />
       
      <div
        class="title"
      >
        Keine Professoren
      </div>
       
      <div
        class="line"
      />
       
      <div
        class="info"
      >
        <p>
          Es wurde kein Professor gefunden.
        </p>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Professors renders correctly 1`] = `
<div
  class="wrapper"
>
  <nav>
    <div
      class="container"
    >
      <label>
        <fa-icon-stub
          class="icon icon-prepend icon-search"
          icon="search"
        />
         
        <input
          class="search"
          placeholder="page.professors.search"
        />
      </label>
    </div>
  </nav>
   
  <transition-stub
    mode="out-in"
    name="fade"
  >
    <div
      class="wrapper"
    >
      <div
        class="list"
      >
        <div
          class="container"
        >
          <transition-group-stub
            name="list-complete"
            tag="p"
          >
            <div
              class="professor article"
            >
              <div
                class="title"
              >
                Prof. Dr. Rick Sanchez
              </div>
               
              <div
                class="data"
              >
                <div
                  class="infos"
                >
                  <div>
                    <fa-icon-stub
                      class="room icon-prepend icon-fw icon"
                      icon="map-marker-alt"
                    />
                    
                    3/232
                  
                  </div>
                   
                  <div>
                    <fa-icon-stub
                      class="time icon-prepend icon-fw icon"
                      icon="calendar-alt"
                    />
                    
                    Donnerstag 11:20 - 12:20
                  
                  </div>
                </div>
                 
                <div
                  class="actions"
                >
                  <a
                    class="icon icon-phone green"
                    href="tel:+49 (0)711 1245 8901"
                  >
                    <fa-icon-stub
                      icon="phone"
                    />
                  </a>
                   
                  <a
                    class="icon icon-envelope blue"
                    href="mailto:rick.sanchez@hft-stuttgart.de"
                  >
                    <fa-icon-stub
                      icon="envelope"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="professor article"
            >
              <div
                class="title"
              >
                Prof. Dr.-Ing. Morty Smith
              </div>
               
              <div
                class="data"
              >
                <div
                  class="infos"
                >
                  <div>
                    <fa-icon-stub
                      class="room icon-prepend icon-fw icon"
                      icon="map-marker-alt"
                    />
                    
                    1/201
                  
                  </div>
                   
                  <div>
                    <fa-icon-stub
                      class="time icon-prepend icon-fw icon"
                      icon="calendar-alt"
                    />
                    
                    nach Vereinbarung
                  
                  </div>
                </div>
                 
                <div
                  class="actions"
                >
                  <a
                    class="icon icon-phone green"
                    href="tel:+49 (0)711 1245 8910"
                  >
                    <fa-icon-stub
                      icon="phone"
                    />
                  </a>
                   
                  <a
                    class="icon icon-envelope blue"
                    href="mailto:morty.smith@hft-stuttgart.de"
                  >
                    <fa-icon-stub
                      icon="envelope"
                    />
                  </a>
                </div>
              </div>
            </div>
          </transition-group-stub>
        </div>
      </div>
       
      <div
        class="note"
      >
        <div
          class="container"
        >
          <p>
            page.professors.title
          </p>
           
          <p>
            
            © 
            <a
              href="http://www.hft-stuttgart.de/"
              target="_blank"
            >
              Hochschule für Technik Stuttgart
            </a>
          </p>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
